.github/actions/docker-image/action.yml

```diff
@@ -44,10 +44,13 @@ runs:
         cache-to: type=gha,mode=max
         outputs: type=image,name=${{ inputs.images }},push-by-digest=true,name-canonical=true,push=true
 
+    # This will upload the digest for each architecture to the same artifact,
+    # Note: upload-artifact@v4 will no longer support this.
+    # https://github.com/actions/upload-artifact?tab=readme-ov-file#not-uploading-to-the-same-artifact
     - name: Export digest
       shell: bash
       run: |
-        digest_dir="${{ runner.temp }}/${{ inputs.target }}-digests"
+        digest_dir="${{ runner.temp }}/${{ inputs.target }}-${{ github.run_number }}-digests"
```

Comment:
Just for my understanding, if `temp` dir doesnt cleanup after every job, wouldnt this overtime accumulate lots of digests with each run? If that is the case, do you think its instead better if we run a cleanup step before the `Build and push docker` step where we clean the `temp` dir? What are your thoughts on the same? I was reading that some teams are cleaning up the whole github workspace [here](https://github.com/orgs/community/discussions/51329)

Reply:
https://github.com/EspressoSystems/nitro-espresso-integration/pull/182/commits/ad339ce4b86e9f4060e19f25a43aeedc9a130267